File: Behavioral/Callback/java/CallbackDemo.java
interface CallbackHandler {
    void onCallback();
}

public class Main {
    public static void main(String[] args) {
        System.out.println("calling a function with callback defined");
        myFunction(new CallbackHandler() {
            @Override
            public void onCallback() {
                System.out.println("got callback");
            }
        });
    }

    static void myFunction(CallbackHandler callbackHandler) {
        // do something ...
        System.out.println("function called");

        // callback
        callbackHandler.onCallback();
    }
}

File: Behavioral/Callback/python/callback_demo.py
def callback_fun():
    print("got callback")
    pass


def my_fun(callback):
    print("function called")
    callback()
    pass


def main():
    print("call a function with callback")
    my_fun(callback_fun)
    pass


if __name__ == "__main__":
    main()
